lib/std/simd.zig

```diff
@@ -278,40 +278,113 @@ test "vector shifting" {
     try std.testing.expectEqual([4]u32{ 40, 30, 20, 10 }, reverseOrder(base));
 }
 
+/// Return the index of the first true bool in `vec` or null if there
+/// is none.
 pub fn firstTrue(vec: anytype) ?VectorIndex(@TypeOf(vec)) {
     const len = vectorLength(@TypeOf(vec));
-    const IndexInt = VectorIndex(@TypeOf(vec));
 
-    if (!@reduce(.Or, vec)) {
-        return null;
+    // FIXME: add other backends when they implement bit-casting the same way
+    if (comptime builtin.cpu.arch.isX86() and builtin.zig_backend == .stage2_llvm) {
+        // NOTE: if the semantics of bit casting bool vector to int
+        // ever change or are formally specified, update this code
```

Comment:
Every comment is implicitly a "note".

```suggestion
        // If the semantics of bit casting bool vector to int
        // ever change or are formally specified, update this code
```

Reply:
Done.